src/logging.c

```diff
@@ -86,6 +87,55 @@ rcutils_ret_t rcutils_logging_initialize(void)
   return rcutils_logging_initialize_with_allocator(rcutils_get_default_allocator());
 }
 
+enum rcutils_get_env_retval
+{
+  RCUTILS_GET_ENV_ERROR = -1,
+  RCUTILS_GET_ENV_ZERO = 0,
+  RCUTILS_GET_ENV_ONE = 1,
+  RCUTILS_GET_ENV_EMPTY = 2,
+};
+
+// A utility function to get zero or one from an environment variable.
+// Returns RCUTILS_GET_ENV_ERROR if we failed to get the environment variable
+// or if it was something we don't understand.
+// Return RCUTILS_GET_ENV_ZERO if the value in the environment variable is "0",
+// RCUTILS_GET_ENV_ONE if the value in the environment variable is "1", or
+// RCUTILS_GET_ENV_EMPTY if the environment variables is empty.
+static enum rcutils_get_env_retval rcutils_get_env_var_zero_or_one(
+  const char * name, const char * zero_semantic,
+  const char * one_semantic)
+{
+  const char * env_var_value = NULL;
+  const char * ret_str = rcutils_get_env(name, &env_var_value);
+  if (NULL != ret_str) {
+    fprintf(
```

Comment:
Please use our fwrite macro instead, as some users will want to scrub use of `fprintf`:

https://github.com/ros2/rcutils/blob/32486f43f36e9fb09e103c2d10345d876161ea17/include/rcutils/error_handling.h#L45

I realize other parts of this code use `fprintf`, but we're trying to curb that in new code. The use case is to be able to avoid using the stdout or stderr file handles by changing as few lines of code as possible. This is needed if, for safety reasons, you cannot use the system file handles, or on platforms that do not provide a console.

If you still want to use string formatting, you can combine fwrite with `rcutils_snprintf`, like here:

https://github.com/ros2/rcutils/blob/32486f43f36e9fb09e103c2d10345d876161ea17/include/rcutils/error_handling.h#L215-L222

Reply:
So, there's a lot of uses of `fprintf` in this file.  I guess my question is whether we should switch them all?  If that is the case, then I'd suggest a follow-up PR for all of them.

Reply:
I think it would be nice to avoid extending the problem, but you're right that the other calls should be changed at some point and that work is out of scope for this pr. I'd still vote for not adding fprintf's, but again, I'm not going to block this pr on that.